src/test/java/org/opentripplanner/middleware/utils/ItineraryUtilsTest.java

```diff
@@ -0,0 +1,377 @@
+            .collect(Collectors.toList());
+
+        List<Itinerary> processedItineraries = ItineraryUtils.getSameDayItineraries(itineraries, trip, QUERY_DATE);
+        Assertions.assertEquals(2, processedItineraries.size());
+        Assertions.assertTrue(processedItineraries.contains(itineraries.get(0)));
+        Assertions.assertTrue(processedItineraries.contains(itineraries.get(1)));
+    }
+
+    /**
+     * Helper method to create a bare-bones itinerary with start or end time.
+     */
+    private Itinerary simpleItinerary(Long startTime, boolean isArrival) {
+        Itinerary itinerary = new Itinerary();
+        Date date = Date.from(Instant.ofEpochMilli(startTime));
+        if (isArrival) {
+            itinerary.endTime = date;
+        } else {
+            itinerary.startTime = date;
+        }
+        itinerary.legs = new ArrayList<>();
+        return itinerary;
+    }
+
+    /**
+     * Helper method to create a trip with locations, time, and queryParams populated.
+     */
+    private MonitoredTrip makeTestTrip(boolean arriveBy) throws URISyntaxException {
+        Place targetPlace = new Place();
+        targetPlace.lat = 33.80;
+        targetPlace.lon = -84.70; // America/New_York
+
+        Place dummyPlace = new Place();
+        dummyPlace.lat = 33.90;
+        dummyPlace.lon = 0.0; // Africa/Algiers.
+
+        MonitoredTrip trip = new MonitoredTrip();
+        trip.id = "Test trip";
+        trip.queryParams = BASE_QUERY;
+        trip.tripTime = QUERY_TIME;
+
+        if (arriveBy) {
+            trip.from = dummyPlace;
+            trip.to = targetPlace;
+
+            Map<String, String> baseQueryParams = trip.parseQueryParams();
+            baseQueryParams.put("arriveBy", "true");
+            trip.queryParams = ItineraryUtils.toQueryString(baseQueryParams);
+        } else { // departBy
+            trip.from = targetPlace;
+            trip.to = dummyPlace;
+
+            trip.queryParams = BASE_QUERY;
+        }
+        return trip;
+    }
+
+    /**
+     * Data structure for the date check test.
+     */
+    private static class GetDatesTestCase {
+        public final boolean checkAllDays;
+        public final List<String> dates;
+
+        public GetDatesTestCase(boolean checkAllDays, List<String> dates) {
+            this.checkAllDays = checkAllDays;
+            this.dates = dates;
+        }
+    }
+
+    /**
+     * Data structure for the same-day test.
+     */
+    private static class SameDayTestCase {
+        public final boolean isArrival;
+        public final boolean shouldBeSameDay;
+        public final String timeOfDay;
+        public final Long tripTime;
+
+        public SameDayTestCase(String timeOfDay, Long tripTime, boolean isArrival, boolean shouldBeSameDay) {
+            this.isArrival = isArrival;
+            this.shouldBeSameDay = shouldBeSameDay;
+            this.timeOfDay = timeOfDay;
+            this.tripTime = tripTime;
```

Comment:
Please always name a variable to have the specific time units the variable is supposed to represent. Consider naming this `tripTargetTimeEpochMillis`.